### PR TITLE
Add option to request importation of specific domains

### PR DESCRIPTION
**Pull request by kienanstewart**
Useful for debugging specific renewal issues

### CODE REVIEW

**Comment by kienanstewart** [comment]
This PR is ready to be reviewed

**Comment by kienanstewart** [comment]
There's one more thing that should be added here, this code path doesn't change the uid (using alternc's su) before the import. I'm not sure what the impact of that is, but I think it should use the su to match what the other code path is doing

**Comment by kienanstewart** [comment]
I've added the su() for specific certificate importations.

File: src/usr/lib/alternc/generate_certbot.php [hunk 1]
```diff
@@ -44,6 +50,13 @@
 // isn't set, use the default value.
 $REQUEST_CERTS= in_array('certificates', array_keys($options)) ? $options['certificates'] : (
     in_array('c', array_keys($options)) ? $options['c'] : $REQUEST_CERTS);
+$domains = in_array('domains', array_keys($options)) ? $options['domains'] : (
+    in_array('d', array_keys($options)) ? $options['d'] : NULL);
+if ($domains !== NULL) {
+    $domains = explode(',', $domains);
+    $REQUEST_CERTS = 'sepcific';
```
Comment (camlafit) [inline src/usr/lib/alternc/generate_certbot.php:57]: Typo please correct :)

**Review by camlafit — CHANGES_REQUESTED** [review]

File: src/usr/lib/alternc/generate_certbot.php [hunk 2]
```diff
@@ -91,7 +104,50 @@ function vprint( $message, $params ){
...
+                $certificate['fqdn'] == $ssl->default_certificate_fqdn) {
+                unset($certs[$index]);
+            }
+        }
+
+        // Request if there are no valid certificates other than the defaults removed,
+        // or if the request is forced.
+        if (sizeof($certs) <= 0  || $force_request) {
+            vprint(_("Requesting import of certificate for '%s'"), array($domain));
+            $sub_domain = get_sub_domain_id_and_member_by_name($domain);
```
Comment (camlafit) [inline src/usr/lib/alternc/generate_certbot.php:136]: it's a static method from m_dom

**Review by camlafit — CHANGES_REQUESTED** [review]

**Comment by kienanstewart** [comment]
Hi @camlafit, thanks for the review of the proposed changes. I've fixed the elements you've highlighted. Please let me know if there's anything else.

thanks!

File: src/usr/lib/alternc/generate_certbot.php [hunk 3]
```diff
@@ -91,7 +104,50 @@ function vprint( $message, $params ){
...
+        // or if the request is forced.
+        if (sizeof($certs) <= 0  || $force_request) {
+            vprint(_("Requesting import of certificate for '%s'"), array($domain));
+            $sub_domain = m_dom::get_sub_domain_id_and_member_by_name($domain);
+            if ($sub_domain === FALSE) {
+                vprint(_("Failed to find sub_domain id and member by name for '%s'"), array($domain));
+                continue;
+            }
+            $mem->su($sub_domain['member_id']);
+            $certbot->import($domain);
```
Comment (camlafit) [inline src/usr/lib/alternc/generate_certbot.php:142]: add force option 
following #74

Comment (camlafit) [reply src/usr/lib/alternc/generate_certbot.php:142]: Hello

#74 is backported to master; could you update your code to integrate force mode ?

**Pull request closed without merging**